lib/msal-react/src/MsalProvider.tsx

```diff
@@ -3,91 +3,50 @@ import { useContext } from "react";
 import {
     IPublicClientApplication,
     AccountInfo,
-    AuthenticationResult,
-    AuthorizationUrlRequest,
-    EndSessionRequest,
+    BroadcastEvent, BroadcastMessage
 } from "@azure/msal-browser";
 import { MsalContext, IMsalContext } from "./MsalContext";
 
 export type MsalProviderProps = {
     instance: IPublicClientApplication;
 };
 
-export const MsalProvider: React.FunctionComponent<MsalProviderProps> = ({
-    instance,
-    children,
-}) => {
+export const MsalProvider: React.FunctionComponent<MsalProviderProps> = ({instance, children}) => {
     // State hook to store accounts
     const [accounts, setAccounts] = React.useState<AccountInfo[]>(
         // TODO: Remove the `|| []` hack when PR is finally merged to msal/browser
         instance.getAllAccounts() || []
     );
 
     // Callback to update accounts after MSAL APIs are invoked
-    const updateContextState = React.useCallback(() => {
+    const updateContextState = () => {
         // TODO: Remove the `|| []` hack when PR is finally merged to msal/browser
         setAccounts(instance.getAllAccounts() || []);
-    }, [instance]);
-
-    // Wrapped instance of MSAL that updates accounts after MSAL APIs are invoked
-    const wrappedInstance = React.useMemo<IPublicClientApplication>(() => {
-        return {
-            acquireTokenPopup: instance.acquireTokenPopup.bind(instance),
-            acquireTokenRedirect: instance.acquireTokenRedirect.bind(instance),
-            acquireTokenSilent: instance.acquireTokenSilent.bind(instance),
-            getAllAccounts: instance.getAllAccounts.bind(instance),
-            getAccountByUsername: instance.getAccountByUsername.bind(instance),
-            handleRedirectPromise: async (): Promise<AuthenticationResult | null> => {
-                const response = await instance.handleRedirectPromise.call(
-                    instance
-                );
-
-                if (response) {
-                    updateContextState();
-                }
-
-                return response;
-            },
-            loginPopup: async (
-                request: AuthorizationUrlRequest
-            ): Promise<AuthenticationResult> => {
-                const response = await instance.loginPopup.call(
-                    instance,
-                    request
-                );
+    };
+
+    React.useEffect(() => {
+        instance.addEventCallback((message: BroadcastMessage) => {
+            const eventTypes = [
+                BroadcastEvent.LOGIN_SUCCESS,
+                BroadcastEvent.ACQUIRE_TOKEN_SUCCESS,
+                BroadcastEvent.HANDLE_REDIRECT_SUCCESS,
+                BroadcastEvent.SSO_SILENT_SUCCESS,
+                BroadcastEvent.LOGOUT_SUCCESS
+            ];
+    
+            if (eventTypes.indexOf(message.type) > -1) {
```

Comment:
I think a switch statement would be cleaner here

Reply:
I think once we add a 2nd thing this callback needs to do that would make sense (error handling maybe?)

Reply:
done